test/asioexec/test_completion_token.cpp

```diff
@@ -329,16 +330,33 @@ namespace {
     const Executor& ex,
     std::shared_ptr<void>& ptr,
     CompletionToken&& token) {
+    //  This function is seemingly fragile?
+    //
+    //  In release mode slightly different arrangements produce bizarre outcomes:
+    //
+    //  /root/stdexec/test/asioexec/test_completion_token.cpp:380: FAILED:
+    //    CHECK( !ex )
+    //  with expansion:
+    //    false
+    //
+    //  /root/stdexec/test/asioexec/test_completion_token.cpp:382: FAILED:
+    //    CHECK( ptr.use_count() == 1U )
+    //  with expansion:
+    //    -1564306560 == 1
```

Comment:
is this a bug in ASIO then?

Reply:
I chased this around a bit before restructuring the function and realizing that fixed it. The closest I've come to explaining it is that it seems like either a code gen bug, or a bug in `std::shared_ptr`.

The fact that it only surfaced in a release build, not debug build, helps push me in this direction. But walking through a release with debug info build I saw the use count move unexpectedly to a nonsense value, I could never isolate whereat that actually happened.